tsdoc/src/nodes/DocXmlElement.ts

```diff
@@ -0,0 +1,336 @@
+        this._startTagClosingDelimiterExcerpt = new DocExcerpt({
+          configuration: this.configuration,
+          excerptKind: ExcerptKind.XmlStartTag_ClosingDelimiter,
+          content: parameters.startTagClosingDelimiterExcerpt
+        });
+      }
+      if (parameters.endTagOpeningDelimiterExcerpt) {
+        this._endTagOpeningDelimiterExcerpt = new DocExcerpt({
+          configuration: this.configuration,
+          excerptKind: ExcerptKind.XmlEndTag_OpeningDelimiter,
+          content: parameters.endTagOpeningDelimiterExcerpt
+        });
+      }
+      if (parameters.endTagClosingDelimiterExcerpt) {
+        this._endTagClosingDelimiterExcerpt = new DocExcerpt({
+          configuration: this.configuration,
+          excerptKind: ExcerptKind.XmlEndTag_ClosingDelimiter,
+          content: parameters.endTagClosingDelimiterExcerpt
+        });
+      }
+      if (parameters.spacingBetweenStartTagAndChildExcerpt) {
+        this._spacingBetweenStartTagAndChildrenExcerpt = new DocExcerpt({
+          configuration: this.configuration,
+          excerptKind: ExcerptKind.Spacing,
+          content: parameters.spacingBetweenStartTagAndChildExcerpt
+        });
+      }
+      if (parameters.spacingAfterEndTagExcerpt) {
+        this._spacingAfterEndTagExcerpt = new DocExcerpt({
+          configuration: this.configuration,
+          excerptKind: ExcerptKind.Spacing,
+          content: parameters.spacingAfterEndTagExcerpt
+        });
+      }
+    } else {
+      this._name = parameters.name;
+      this._spacingAfterName = parameters.spacingAfterName;
+    }
+
+    for (const xmlAttribute of parameters.xmlAttributes ?? []) {
+      this._xmlAttributes.push(xmlAttribute);
+      this._xmlAttributesByName.set(xmlAttribute.name, xmlAttribute);
+    }
+
+    this._selfClosingTag = !!parameters.selfClosingTag;
+  }
+
+  public get kind(): string {
+    return DocNodeKind.XmlElement;
+  }
+
+  /**
+   * The XML element name.
+   */
+  public get name(): string {
+    if (this._name === undefined) {
+      this._name = this._nameExcerpt!.content.toString();
+    }
+    return this._name;
+  }
+
+  /**
+   * The XML attributes belonging to this XML element.
+   */
+  public get xmlAttributes(): ReadonlyArray<DocXmlAttribute> {
+    return this._xmlAttributes;
+  }
+
+  /**
+   * Attempts to find an XML attribute with the specified name.
+   * @param name - The name of the XML attribute to get.
+   * @returns The XML attribute with the specified name, or `undefined` if no such attribute with the given name exists.
+   */
+  public tryGetXmlAttribute(name: string): DocXmlAttribute | undefined {
+    if (!DocXmlElement._isValidXmlName(name)) {
+      throw new Error(`'${name}' is not a valid XML attribute name.`);
+    }
+
+    return this._xmlAttributesByName.get(name);
+  }
+
+  private static _isValidXmlName(name: string): boolean {
+    return /^[a-zA-Z_][a-zA-Z0-9_.-]*$/.test(name);
```

Comment:
`.` should not be allowed in attribute names.  At least this wasn't permitted in the previous doc, and the parser doesn't accept it because it relies on `StringChecks.explainIfInvalidXmlName()`.  

(The specs for XML/HTML/CommonMark do allow many other unusual characters, but for TSDoc we've taken a minimalist approach of only allowing syntaxes with a compelling justification. This is partly to make it more intuitive for humans to predict how TSDoc will get parsed, and partly to avoid difficult ambiguity/escaping edge cases caused by combining grammars of JSDoc, XML, and markdown.)

Reply:
I will replace this RegExp with `StringChecks.explainIfInvalidXmlName()`